### PR TITLE
Omit duplicate wf error in first invocation & not found in cleanup

### PR DESCRIPTION
**Pull request by CahidArda**
There are cases where network errors cause workflow runs to fail.

This PR handles three of those cases:
1. when the workflow is first created, it is possible for request to fail even though it is delivered to QStash. So QStash creates the workflow but the SDK retries. Solution is to omit this error.
2. when the SDK attempts to delete the workflow run after it completes, the request may fail even though it was delivered to QStash. Solution is to omit this error if it occurs.
3. The workflow endpoint can get an error after scheduling the next step. In this case, the scheduled request may complete the workflow and finish it while the request (which failed) retries after the workflow is finished. We omit the workflow-canceled error to fix this issue.

In both cases, the debugger will log warnings.

### CODE REVIEW

**Comment by CahidArda** [comment]
this won't be merged right away. We will get a canary release from this branch to verify the solution. If all looks good, we will merge to main.

**Comment by CahidArda** [comment]
we should update client.trigger in this PR too https://github.com/upstash/workflow-js/pull/18

File: src/workflow-requests.ts [hunk 1]
```diff
@@ -94,16 +122,37 @@ export const triggerWorkflowDelete = async <TInitialPayload>(
...
+    await debug?.log(
+      "SUBMIT",
+      "SUBMIT_CLEANUP",
+      `workflow run ${workflowContext.workflowRunId} deleted.`
+    );
+    return { deleted: true };
+  } catch (error) {
+    if (
+      error instanceof QstashError &&
+      error.message.includes(`workflowRun ${workflowContext.workflowRunId} not found`)
```
Comment (sancar) [inline src/workflow-requests.ts:145]: I think it is better to rely on the status code.

File: src/workflow-requests.ts [hunk 2]
```diff
@@ -79,7 +97,17 @@ export const triggerRouteFunction = async ({
     return ok("workflow-finished");
   } catch (error) {
     const error_ = error as Error;
-    if (!(error_ instanceof WorkflowAbort)) {
+    if (
+      error instanceof QstashError &&
+      error.message.includes("can not append to a a cancelled workflow")
```
Comment (sancar) [inline src/workflow-requests.ts:102]: Same here. Message might be refactored but status code should be same. We can use BadRequest(400) here. You can print the message on the log for debugging purposes.

**Comment by CahidArda** [comment]
can't pass the tests because they are too flaky. we had passed the tests in [51b71eb](https://github.com/upstash/workflow-js/pull/16/commits/51b71eb44be80d5773077e18f117c700a071da9f), and after that change we only bumped qstash and updated how we check omit error cases.

The unit tests pass. I ran the integration test which fails locally and it works. We can go ahead without the CI.

**Review by fahreddinozcan — APPROVED** [review]

**Pull request merged**